packages/server/socketHandlers/handleUpgrade.ts

```diff
@@ -1,38 +1,14 @@
 import {WebSocketBehavior} from 'uWebSockets.js'
-import {TrebuchetCloseReason} from '../../client/types/constEnums'
-import safetyPatchRes from '../safetyPatchRes'
-import {isAuthenticated} from '../utils/authorization'
-import checkBlacklistJWT from '../utils/checkBlacklistJWT'
 import getQueryToken from '../utils/getQueryToken'
-import sendToSentry from '../utils/sendToSentry'
 import uwsGetIP from '../utils/uwsGetIP'
 
-const handleUpgrade: WebSocketBehavior<void>['upgrade'] = async (res, req, context) => {
-  safetyPatchRes(res)
-  const protocol = req.getHeader('sec-websocket-protocol')
-  if (protocol !== 'trebuchet-ws') {
-    sendToSentry(new Error(`WebSocket error: invalid protocol: ${protocol}`))
-    // WS Error 1002 is roughly HTTP 412 Precondition Failed because we can't support the req header
-    res.writeStatus('412').end()
-    return
-  }
```

Comment:
-1 I think we need to distinguish 2 cases here: The client speaks our language but says the wrong thing (right protocol, wrong auth) and we don't understand it at all (wrong protocol). I think in the latter case, we should not move forward with the upgrade which should also fail the client permanently because it could never connect. Only when the protocol matches, we should upgrade and do the other checks. In that case the client can talk websocket with the server, so there is no need to try other connection methods.